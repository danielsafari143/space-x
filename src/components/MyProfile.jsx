import React from 'react';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const data = useSelector((state) => state.mission);
  return (
    <div>
      <div className="missions">
        {
           data.mission.map((item) => (item.reserved !== undefined && item.reserved ? <p key={item.id}>{item.mission_name}</p> : ''))
         }
      </div>
    </div>
  );
};

export default MyProfile;
