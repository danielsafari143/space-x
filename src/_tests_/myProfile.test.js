import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import axios from 'axios';
import Items from './data';
import MyProfile from '../components/MyProfile';
import missionItem from './missionData';

jest.mock('axios');

const reducer = (state = {
  mission: { mission: missionItem },
  rockets: { rocket: Items },
}) => state;

const store = configureStore({ reducer });

describe('should test MyProfile', () => {
  it('should test the snapshot of main page', () => {
    axios.get.mockResolvedValue({ data: missionItem });
    axios.get.mockResolvedValue({ data: Items });
    const navb = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <MyProfile />
        </Provider>
      </MemoryRouter>,
    ).toJSON();
    expect(navb).toMatchSnapshot();
  });
});
