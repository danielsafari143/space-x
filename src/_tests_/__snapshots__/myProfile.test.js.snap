// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test MyProfile should test the snapshot of main page 1`] = `
<div
  className="myprofile"
>
  <div
    className="missions"
  >
    <h3>
      Missions
    </h3>
  </div>
  <div
    className="rockets"
  >
    <h3>
      Rockets
    </h3>
  </div>
</div>
`;
